LibOS/shim/src/sys/shim_exit.c

```diff
@@ -83,8 +83,9 @@ int thread_exit(struct shim_thread * self, bool send_ipc)
         debug("thread exits, notifying thread %d\n", parent->tid);
 
         lock(parent->lock);
-        list_del_init(&self->siblings);
-        list_add_tail(&self->siblings, &parent->exited_children);
+        listp_del_init(self, &parent->children, siblings);
```

Comment:
This is an especially tricky one. There is no condition to tell whether the thread is in children or exit_children.